### ISSUE TITLE
Eliminate fingerprinting in Prometheus TSDB input for Elastic 8.13 and above

### ISSUE THREAD

**Issue by StephanErb** [elastic/integrations#9400]
The current Prometheus integration relies on a fingerprint dimension:

```
    - name: labels_fingerprint
      type: keyword
      dimension: true
      description: Autogenerated ID representing the fingerprint of labels object and includes query name
```

Now that https://github.com/elastic/elasticsearch/issues/93564 is resolved in 8.13, this is no longer needed. Instead, `labels.*` can be marked as dimensions directly. Due to the locality sensitive hashing used in https://github.com/elastic/elasticsearch/pull/98023 we can expect that https://github.com/elastic/elasticsearch/pull/99747 will deliver storage savings over the current approach. 

cc @felixbarny  @tetianakravchenko

**Comment by felixbarny** [comment]
I agree this is unblocked now und would lead to better performance and storage reduction. @lalit-satapathy do you think we can prioritize this?

**Comment by lalit-satapathy** [comment]
Hi @felixbarny,

Do we know if now [objects](https://github.com/elastic/integrations/blob/main/packages/prometheus/data_stream/collector/fields/fields.yml#L5) can be marked dimension now ?  Dont see them in the [dimension](https://www.elastic.co/guide/en/elasticsearch/reference/current/tsds.html#time-series-dimension) doc.

**Comment by felixbarny** [comment]
We don't even need that because we can just create a dynamic template for `prometheus.labels.*` that defines labels as a keyword field type with `time_series_dimension` set to `true`.

Something like this:

```json
{
  "mappings": {
    "dynamic_templates": [
      {
        "labels": {
          "path_match": "prometheus.labels.*",
          "mapping": {
            "type": "keyword",
            "time_series_dimension": true
          }
        }
      }
    ]
  }
}
```

**Comment by tetianakravchenko** [comment]
For the remote_write datastream: as I understand it will be still required to keep the fingerprint processor for the list of metric_names - https://github.com/elastic/integrations/blob/main/packages/prometheus/data_stream/remote_write/elasticsearch/ingest_pipeline/default.yml#L9-L24

but for [collector](https://github.com/elastic/integrations/blob/main/packages/prometheus/data_stream/collector/elasticsearch/ingest_pipeline/default.yml) - all ingest pipeline can be removed
and for [query](https://github.com/elastic/integrations/blob/main/packages/prometheus/data_stream/query/elasticsearch/ingest_pipeline/default.yml)  - fingerprint processor can be removed

**Comment by felixbarny** [comment]
> For the remote_write datastream: as I understand it will be still required to keep the fingerprint processor for the list of metric_names

That's right. But we could remove the fingerprinting of labels and only do a fingerprint of the metric names.

> I am getting an error when building package:

Seems like there's a gap in the support in elastic-package and Fleet to support the dimension parameter for dynamic templates. But Elasticsearch and TSDB does support that.  Could you create an issue for this? It seems like it should be an easy fix.

**Comment by lalit-satapathy** [comment]
> Seems like there's a gap in the support in elastic-package and Fleet to support the dimension parameter for dynamic templates.

Yes, we need to mostly would need to go through the fleet enhancements for this. We did exactly the same for dynamic templates for _metric types_ [here](https://github.com/elastic/kibana/issues/155004).

**Comment by tetianakravchenko** [comment]
fleet issue - https://github.com/elastic/kibana/issues/179649

@felixbarny @lalit-satapathy can you please forward it to the correct team?

**Comment by jsoriano** [comment]
> adding `dimension: true` to the definition above should produce the expected dynamic template, but it is not supported now. When using:
> 
> ```
>     - name: labels.*
>       type: object
>       object_type: keyword
>       dimension: true
>       description: |
>         Prometheus metric labels
> ```
> I am getting an error when building package:
> 
> ```
> elastic-package-0.98.2 build -v
> ...
>    1. file ".../go/src/github.com/elastic/integrations/build/packages/prometheus-1.16.0.zip/data_stream/collect
> ```

@tetianakravchenko please try with the "compact" syntax:
```
    - name: labels.*
      type: keyword
      dimension: true
      description: |
        Prometheus metric labels
```

**Comment by lalit-satapathy** [comment]
Yes, we can take this up as the fleet [issue](https://github.com/elastic/kibana/issues/179649) is resolved. CC: @gpop63

**Comment by lalit-satapathy** [comment]
Since both testing by @gpop63 [here](https://github.com/elastic/integrations/pull/9785#issuecomment-2110939045) and @harnish-elastic [here](https://github.com/elastic/integrations/pull/9785#issuecomment-2112483492) has confirmed that the new capability only works 8.14 onwards, we should enable the same in packages stack dependancy.

FYI: The relevant Fleet [change](https://github.com/elastic/kibana/pull/180023) has gone in 8.14.0.

**Comment by lalit-satapathy** [comment]
@agithomas will be creating a separate meta issue to track other packages need such migration, which can be taken in the future. Please link the issue here.

**Comment by agithomas** [comment]
> @agithomas will be creating a separate meta issue to track other packages need such migration, which can be taken in the future. Please link the issue here.

@lalit-satapathy , please find the link to the [meta issue](https://github.com/elastic/integrations/issues/9910)

**Comment by gpop63** [comment]
@felixbarny we are fine with removing the `labels_fingerprint` fields from these data streams even if they are in GA?

**Comment by felixbarny** [comment]
As this is just an internal field that doesn't really provide value by itself, other than aiding the enablement of TSDB, I think that's fine.
